File: python/element_appearing_more_than_25%_in_sorted_array.py
# no. 1287
class Solution:
    def findSpecialInteger(self, arr: List[int]) -> int:
        if len(arr) == 1:
            return arr[0]
        threshold = len(arr) // 4
        count = 1
        curr = arr[0]
        for i in range(1, len(arr)):
            if arr[i] == curr:
                count += 1
            else:
                curr = arr[i]
                count = 1
            if count > threshold:
                return curr

File: C++/element_appearing_more_than_25%_in_sorted_array.cpp
//no. 1287
#include<vector>
using namespace std;
class Solution {
public:
    int findSpecialInteger(vector<int>& arr) {
        if(arr.size() == 1){
            return arr[0];
        }
        int threshold = arr.size() / 4;
        int count = 1;
        int curr = arr[0];
        for(int i = 1; i < arr.size(); i++){
            if(arr[i] == curr){
                count++;
            }else{
                curr = arr[i];
                count = 1;
            }
            if(count > threshold){
                return curr;
            }
        }
        return 0;
    }
};
